docs/microphone.rst

```diff
@@ -70,11 +117,59 @@ Functions
     * **return**: a representation of the sound pressure level in the range 0 to
       255.
 
+.. py:function:: record(duration=3000, rate=7812)
+
+    Record sound into an ``AudioFrame`` for the amount of time indicated by
+    ``duration`` at the sampling rate indicated by ``rate``.
+
+    The amount of memory consumed is directly related to the length of the
+    recording and the sampling rate. The higher these values, the more memory
+    it will use.
+
+    A lower sampling rate will reduce both memory consumption and sound
+    quality.
+
+    If there isn't enough memory available a ``MemoryError`` will be raised.
+
+    :param duration: How long to record in milliseconds.
+    :param rate: Number of samples to capture per second.
+    :returns: An ``AudioFrame`` with the sound samples.
+
+.. py:function:: record_into(buffer, rate=7812, wait=True)
 
-Example
-=======
+    Record sound into an existing ``AudioFrame`` until it is filled,
+    or the ``stop_recording()`` function is called.
 
-An example that runs through some of the functions of the microphone API::
+    :param buffer: An ``AudioFrame`` to record sound.
+    :param rate: Number of samples to capture per second.
+    :param wait: When set to ``True`` it blocks until the recording is
+        done, if it is set to ``False`` it will run in the background.
+
+.. py:function:: is_recording()
+
+    :returns: ``True`` if the microphone is currently recording sound, or
+      ``False`` otherwise.
+
+.. py:function:: stop_recording()
+
+    Stops an a recording running in the background.
```

Comment:
```suggestion
    Stops a recording running in the background.
```

Fixing typo but that also got me wondering what happens if you record more than once with wait=False (before the first has finished). 
Do they all work? If so does this function stop them all?

Reply:
Only one can be recorded at once, so a previously running recording is aborted at the point you call `record()` or `record_into()` again (as though you called `stop_recording()` first).

Reply:
This issue to discuss some implementation enhancements would cover this as well:
- https://github.com/microbit-foundation/micropython-microbit-v2/issues/198